javatests/arcs/core/host/HandleAdapterTest.kt

```diff
@@ -109,13 +110,79 @@ class HandleAdapterTest {
         assertThat(writeOnlyHandle).isNotInstanceOf(ReadWriteCollectionHandle::class.java)
     }
 
+    @Test
+    fun singletonHandleAdapter_onUpdateTest() = runBlockingTest {
+        val handle = manager.createSingletonHandle(
+            HandleMode.ReadWrite,
+            READ_WRITE_HANDLE,
+            Person,
+            STORAGE_KEY,
+            Person.SCHEMA
+        ) as ReadWriteSingletonHandle<Person>
+
+        var x = 0
+        handle.onUpdate { x = 1 }
```

Comment:
maybe also verify that the correct entity came through?

Reply:
You should be able to do:

```
assertThat(it).isEqualTo(Person())
```

(You could set fields on the entity to ensure it's the right one)

Reply:
I did it a slightly different way, but same idea :)